src/Service.GraphQLBuilder/Mutations/CreateMutationBuilder.cs

```diff
@@ -15,22 +15,28 @@ namespace Azure.DataApiBuilder.Service.GraphQLBuilder.Mutations
 {
     public static class CreateMutationBuilder
     {
+        private const string CREATE_MULTIPLE_MUTATION_SUFFIX = "Multiple";
         public const string INPUT_ARGUMENT_NAME = "item";
+        public const string CREATE_MUTATION_PREFIX = "create";
 
         /// <summary>
-        /// Generate the GraphQL input type from an object type
+        /// Generate the GraphQL input type from an object type for relational database.
         /// </summary>
         /// <param name="inputs">Reference table of all known input types.</param>
         /// <param name="objectTypeDefinitionNode">GraphQL object to generate the input type for.</param>
         /// <param name="name">Name of the GraphQL object type.</param>
+        /// <param name="baseEntityName">In case when we are creating input type for linking object, baseEntityName is equal to the targetEntityName,
+        /// else baseEntityName is equal to the name parameter.</param>
         /// <param name="definitions">All named GraphQL items in the schema (objects, enums, scalars, etc.)</param>
-        /// <param name="databaseType">Database type to generate input type for.</param>
+        /// <param name="databaseType">Database type of the relational database to generate input type for.</param>
         /// <param name="entities">Runtime config information.</param>
         /// <returns>A GraphQL input type with all expected fields mapped as GraphQL inputs.</returns>
-        private static InputObjectTypeDefinitionNode GenerateCreateInputType(
+        private static InputObjectTypeDefinitionNode GenerateCreateInputTypeForRelationalDb(
             Dictionary<NameNode, InputObjectTypeDefinitionNode> inputs,
             ObjectTypeDefinitionNode objectTypeDefinitionNode,
+            string entityName,
```

Comment:
What is entityName here? in a relation is this the entityname for the child? This naming is a bit confusing to me. Does maybe baseEntityName and TargetEntityName make more sense?

Reply:
This is the `entityName` for which we want to create an input type. Depending on from where the method is called, it might be a child related entity, or the parent entity. So, just look at this standalone function which takes in an entity and creates an input type for it.

Caller 1: https://github.com/Azure/data-api-builder/blob/fa9bcab5ed8ddaf516f623e02ef20bf86951d294/src/Service.GraphQLBuilder/Mutations/CreateMutationBuilder.cs#L475

Caller2: https://github.com/Azure/data-api-builder/blob/fa9bcab5ed8ddaf516f623e02ef20bf86951d294/src/Service.GraphQLBuilder/Mutations/CreateMutationBuilder.cs#L318